streams/src/test/java/org/apache/kafka/streams/integration/EosBetaUpgradeIntegrationTest.java

```diff
@@ -837,6 +866,11 @@ public void init(final ProcessorContext context) {
                             crash = errorInjectedClient2;
                             sharedCommit = commitCounterClient2;
                         }
+                        punctuator = context.schedule(
```

Comment:
There's going to be a separate punctuator per task, right? Does the test account for this?

Reply:
Sort of; with eos-beta when you need to commit one task, you need to commit all of them, so requesting a commit at all should be sufficient. It's kind of a subtle point, but I think it's actually preferable to just request a commit on one task so this test can also verify that we commit all tasks together

Reply:
Ok, I was thinking more along the lines that we have just one requestCommit instance while we would have multiple punctuator instances all using it. It looked like a bug, but it seems like it'll be fine.

Reply:
In this test we would have multiple punctuator indeed but they would be executed by a single thread sequentially so that's fine.